internals/overlord/state/notices.go

```diff
@@ -81,7 +85,21 @@ type Notice struct {
 }
 
 func (n *Notice) String() string {
-	return fmt.Sprintf("Notice %s (%s:%s)", n.id, n.noticeType, n.key)
+	return fmt.Sprintf("Notice %s (%d:%s:%s)", n.id, n.userID, n.noticeType, n.key)
+}
+
+// UserID returns whether the notice has a user ID, and if so, what the UID is.
+func (n *Notice) UserID() (hasUserID bool, userID uint32) {
+	// Importantly, doesn't expose the address of notice's user ID, so the
+	// value cannot be mutated.
+	return expandUserID(n.userID)
+}
+
+func expandUserID(userID *uint32) (hasUserID bool, uid uint32) {
```

Comment:
I don't think this function really pays for itself, and suggest we just inline it in the two places it's used in this package. Though maybe it just needs a better name? (Can't think of one off the top of my head though.) I don't have strong feelings, though -- feel free to ignore this comment.

Reply:
Would `flattenUserID` be appropriate, as it lets us flatten a `*uint32` into map-key-compatible values?

I also do like using this as a function (as opposed to inline) when constructing the `noticeKey`, as it keeps `AddNotice` much cleaner and keeps a single source for how to parse `userID *uint32` values in that way. I assume the compiler should inline it automatically? Not a huge deal though, I can change it to inline if you'd prefer.

Reply:
That's fine. I like `flattenUserID` -- thanks. (Yep, the compiler will inline small functions either way.)